src/server/models/mediaquery.js

```diff
@@ -3,17 +3,78 @@ const client = require('./db');
 
 const { DB_DATABASE } = process.env;
 
-module.exports.getCastsByCourse = async courseLabel => {
-  const castCollection = client.db(DB_DATABASE).collection('bruincastmedia');
-  const toBeReturned = await castCollection
-    .find({ classShortname: courseLabel })
+module.exports.getCastsByCourse = async (dbCollection, courseLabel) => {
+  /*
+   * Aggregation Steps:
+   * 1. Match by courseLabel
+   * 2. Sort course records by date
+   * 3. Group buckets of records by 'week' field
+   *
+   * Due to the way MongoDB aggregation works in how it buckets these items,
+   * there needs to be another boundary after 'Finals' in order for 'Finals'
+   * items to be grouped together
+   */
+  const aggregation = [
+    {
+      $match: {
+        classShortname: courseLabel,
+      },
+    },
+    {
+      $sort: {
+        date: 1,
+      },
+    },
+    {
+      $bucket: {
+        groupBy: '$week',
+        boundaries: [
+          '0',
+          '1',
+          '10',
+          '2',
+          '3',
+          '4',
+          '5',
+          '6',
+          '7',
+          '8',
+          '9',
+          'Finals',
+          'More',
```

Comment:
@chern What is more? When is this triggered? There isn't a test case to cover this.. so either define a test for this use case or delete it.

Reply:
@rlorenzo The boundaries in aggregation have an inclusive lowerbound and exclusive upperbound; in this case, the 'Finals' range needs an upperbound after it in order for the records with 'Finals' to be grouped, which is why I had added 'More' so that the bounds would be ['Finals', 'More').

I'm updating the code right now to store the week numbers as ints, and I notice that I still need to put another bound after 88 in order for the records with week 88 to be grouped properly, so I'm adding 89 at the end such that the bounds will be [88, 89).

The bounds are explained in the mongo docs example: https://docs.mongodb.com/manual/reference/operator/aggregation/bucket/#examples

Reply:
@chern This should be in the comments so someone reading it understands why you are including a week that doesn't actually exist.